src/app/core/cms/abstract.service.ts

```diff
@@ -0,0 +1,68 @@
+import { Observable, ReplaySubject } from 'rxjs';
+import { Http } from '@angular/http';
+import { HalDoc } from './haldoc';
+import { HalObservable } from './halobservable';
+import { HalRemote } from './halremote';
+
+export abstract class AbstractService {
```

Comment:
CmsService could/should also extend AbstractService (or BaseService or whatever we want to call it)